src/scripts/main.js

```diff
@@ -1,3 +1,223 @@
+  const label = document.createElement('label');
+
+  input.type = type;
+  input.name = nameInput;
+  label.textContent = `${text}:`;
+  input.dataset.qa = dataset;
+
+  label.append(input);
+
+  return label;
+}
+
+function notification(isSuccess, title, message) {
+  const mes = document.createElement('div');
+
+  mes.dataset.qa = 'notification';
+  mes.classList.add('notification');
+
+  if (isSuccess) {
+    mes.classList.add('success');
+  } else {
+    mes.classList.add('error');
+  }
+
+  const titleMes = document.createElement('h3');
+
+  titleMes.textContent = title;
+
+  const messageText = document.createElement('p');
+
+  messageText.textContent = message;
+
+  mes.append(titleMes, messageText);
+
+  document.body.append(mes);
+
+  setTimeout(() => mes.remove(), 3000);
+}
+
+let currentSortColumn = null;
+let isAscending = true;
+
+table.addEventListener('click', (e) => {
+  if (e.target.tagName !== 'TH') {
+    return;
+  }
+
+  const index = e.target.cellIndex;
+  const content = firstRow.cells[index].textContent.trim();
+  const type = !isNaN(content.replace(/[,$]/g, '')) ? 'number' : 'string';
+
+  if (currentSortColumn === index) {
+    isAscending = !isAscending;
+  } else {
+    currentSortColumn = index;
+    isAscending = true;
+  }
+
+  rows.length = 0;
+  rows.push(...body.rows);
+
+  rows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].textContent.trim().replace(/[,$]/g, '');
+    const cellB = rowB.cells[index].textContent.trim().replace(/[,$]/g, '');
+
+    let result;
+
+    if (type === 'number') {
+      result = parseFloat(cellA) - parseFloat(cellB);
+    } else {
+      result = cellA.localeCompare(cellB);
+    }
+
+    return isAscending ? result : -result;
+  });
+
+  rows.forEach((child) => body.appendChild(child));
+});
+
+table.addEventListener('click', (e) => {
+  if (e.target.tagName !== 'TD') {
+    return;
+  }
+
+  rows.forEach((row) => row.classList.remove('active'));
+
+  const clickedRow = e.target.parentElement;
+
+  clickedRow.classList.add('active');
+});
+
+table.addEventListener('dblclick', (e) => {
+  if (e.target.tagName !== 'TD') {
+    return;
+  }
+
+  if (currentEditingCell) {
+    return;
+  }
+
+  const cell = e.target;
+  const initialValue = cell.textContent.trim();
+  const input = document.createElement('input');
+
+  input.value = initialValue;
+  input.classList.add('cell-input');
+
+  cell.textContent = '';
+  cell.appendChild(input);
+
+  input.focus();
+
+  input.addEventListener('keydown', (ev) => {
+    if (ev.key === 'Enter') {
+      saveChanges(input, cell, initialValue);
+    }
+  });
+
+  input.addEventListener('blur', () => {
+    saveChanges(input, cell, initialValue);
+  });
+
+  currentEditingCell = cell;
+});
+
+function saveChanges(input, cell, initialValue) {
+  const newValue = input.value.trim();
+
+  if (newValue === '') {
+    cell.textContent = initialValue;
+  } else {
+    cell.textContent = newValue;
+  }
```

Comment:
In the `saveChanges` function, ensure that `currentEditingCell` is reset to `null` even if the new value is empty. This will prevent the `currentEditingCell` from being incorrectly set if the user leaves the input empty.